### PR TITLE
datadog-serverless: fixed metrics and logs ingestion from Datadog serverless extensions for AWS and GCP

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
fixes https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7761

### Describe Your Changes

- datadog /api/v2/logs api supports message field in json format, which is not documented and is used by serverless extension. This PR allows message field to be both string and object type. Also added support of not documented timestamp field
- added `-datadog.streamFields` and `-datadog.ignoreFields` flags to configure default stream fields for datadog logs, where there's no alternative option to pass extra headers and query args
- added ingest `max` and `min` values of data, which are ingested using `datadogsketches` API, which is also actively used by serverless extensions
- use default `.` separator instead of `_` for sketches metric names until metrics are not sanitized

### Checklist

The following checks are **mandatory**:

- [ ] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: app/vlinsert/datadog/datadog.go [hunk 1]
```diff
@@ -129,19 +142,43 @@ func readLogsRequest(ts int64, data []byte, lmp insertutils.LogMessageProcessor)
...
-				return
-			}
-			switch string(k) {
+			switch bytesutil.ToUnsafeString(k) {
 			case "message":
-				fields = append(fields, logstorage.Field{
-					Name:  "_msg",
-					Value: bytesutil.ToUnsafeString(val),
-				})
+				switch v.Type() {
```
Comment (f41gh7) [inline app/vlinsert/datadog/datadog.go:147]: I suggest to add a new function:

```go
// datadog message field has two formats:
//   - regular log message with string text
//   - nested json format for serverless plugins
//     which has folowing format:
//     {"message": {"message": "text","lamdba": {"arn": "string","requestID": "string"}} }
//
// See https://github.com/DataDog/datadog-agent/blob/1c318b7c0025332108f8d45192ffd6069b5331b5/pkg/logs/processor/json_serverless.go#L25
func appendMsgFields(fields []logstorage.Field, v *fastjson.Value) ([]logstorage.Field, error) {
	switch v.Type() {
	case fastjson.TypeString:
		val := v.GetStringBytes()
		fields = append(fields, logstorage.Field{
			Name:  "_msg",
			Value: bytesutil.ToUnsafeString(val),
		})
	case fastjson.TypeObject:
		var firstErr error
		v.GetObject().Visit(func(k []byte, v *fastjson.Value) {
			if firstErr != nil {
				return
			}
			switch string(k) {
			case "message":
				val := v.GetStringBytes()
				fields = append(fields, logstorage.Field{
					Name:  "_msg",
					Value: bytesutil.ToUnsafeString(val),
				})
			case "lamdba":
				obj, err := v.Object()
				if err != nil {
					firstErr = err
					firstErr = fmt.Errorf("unexpected lambda value type for %q:%q; want object", k, v)
					return
				}
				obj.Visit(func(k []byte, v *fastjson.Value) {
					if firstErr != nil {
						return
					}
					val, err := v.StringBytes()
					if err != nil {
						firstErr = fmt.Errorf("unexpected lambda label value type for %q:%q; want string", k, v)
						return
					}
					fields = append(fields, logstorage.Field{
						Name:  bytesutil.ToUnsafeString(k),
						Value: bytesutil.ToUnsafeString(val),
					})
				})

			}
		})
	default:
		return fields, fmt.Errorf("unsupported message type %q", v.Type().String())
	}
	return fields, nil
}
```

 and use it as:

```go
			switch bytesutil.ToUnsafeString(k) {
			case "message":
				fields, err = appendMsgFields(fields, v)
				if err != nil {
					return
				}
```

File: app/vlinsert/datadog/datadog.go [hunk 2]
```diff
@@ -129,19 +142,43 @@ func readLogsRequest(ts int64, data []byte, lmp insertutils.LogMessageProcessor)
...
+					})
+				case fastjson.TypeObject:
+					p := logstorage.GetJSONParser()
+					if e := p.ParseLogMessage(v.MarshalTo(nil)); e != nil {
+						err = fmt.Errorf("cannot parse json-encoded log entry: %w", e)
+						return
+					}
+					logstorage.RenameField(p.Fields, []string{"message"}, "_msg")
+					fields = append(fields, p.Fields...)
+					logstorage.PutJSONParser(p)
```
Comment (f41gh7) [inline app/vlinsert/datadog/datadog.go:162]: given code violates `ParseLogMessage` rule. It holds reference for p.Fields after `PutJSONParser` call.

**Review by f41gh7 — CHANGES_REQUESTED** [review]
Overall looks good. Please see comments

**Comment by AndrewChubatiuk** [comment]
hey @f41gh7 
thanks for review
addressed your comments

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**